src/frequenz/client/reporting/_client.py

```diff
@@ -252,3 +253,33 @@ def dt2ts(dt: datetime) -> PBTimestamp:
         except grpcaio.AioRpcError as e:
             print(f"RPC failed: {e}")
             return
+
+    @staticmethod
+    def calculate_energy_metric(
+        metric_samples: list[MetricSample], time_interval: timedelta
+    ) -> pd.Series:
+        """
+        Calculate the energy metric with AC_ACTIVE_POWER values and time interval.
```

Comment:
I don't remember whether this should have been from power or the actual energy metrics. For latter, the only thing that needs to be done is to remove the resets. That could be achieved by taking the diff, removing the negatives (reset) and cumulatively sum them up again.